src/opnsense/mvc/app/models/OPNsense/Proxy/Proxy.xml

```diff
@@ -239,6 +239,13 @@
               <MaximumValue>32</MaximumValue>
               <ValidationMessage>the number of sslrtd children needs to be an integer value between 1 and 32</ValidationMessage>
             </sslcrtd_children>
+            <snmp_port type="IntegerField">
+                <MinimumValue>0</MinimumValue>
```

Comment:
0 is not valid on a linux/bsd system as far as I know because it means choose a free port in the socket api.

Reply:
If the port number is set to" 0", the SNMP Agent is disabled.

Reply:
While this is how squid does it, I'd rather have an Enable checkbox in the GUI, preferably in its own subtab like FTP proxy and the default port set to 3401 as squid docs state. http://www.squid-cache.org/Doc/config/snmp_port/

We don't have to push every config oddity to the UI.